src/tools/lib/python/particleCalorimeter.py

```diff
@@ -0,0 +1,245 @@
+        `energyHistogram` plugin output. The data is therefore summed
+        over the yaw and pitch angles.
+        """
+        self.hist_data = np.sum(self.calo_data, axis=(1, 2))
+
+    def plot_histogram(self,
+                       with_outliers=False,
+                       norm=False):
+        """
+        Parameters
+        ----------
+        with_outliers : boolean
+            plots also underflow and overflow bins if ``True``
+        norm : boolean
+            normalizes the histogram to the total number of counts
+            in all bins if ``True``
+        """
+        # unit conversion: Joules to eV
+        J2EV = 1.602e-19
+        # unit conversion: simUnit to EV
+        simUnitEV = self.sim_unit / J2EV
+
+        figHist = plt.figure(0)
+        axHist = figHist.add_subplot(111)
+
+        # absolute histogram data [photon count]
+        abs_hist_data = self.hist_data * \
+            simUnitEV / 1e3 / self.energy_range
+        energy_range = self.energy_range
+
+        # remove outliers from data to plot
+        if with_outliers is False:
+            abs_hist_data = abs_hist_data[1:-1]
+            energy_range = energy_range[1:-1]
+
+        if norm is False:
+            axHist.plot(energy_range,
+                        abs_hist_data,
+                        label="{}".format(int(self.step)))
+        else:
+            normed_hist_data = abs_hist_data / np.sum(abs_hist_data)
+            axHist.plot(energy_range,
+                        normed_hist_data,
+                        label="{}".format(int(self.step)))
+
+        axHist.set_xscale("log")
+        axHist.set_yscale("log")
+        axHist.set_xlabel("energy [keV]")
+        axHist.set_ylabel("count [arb. u.]")
+        axHist.set_title("`{}` energy histogram".format(self.species_name))
+        axHist.set_xlim(.5 * self.min_energy, 2 * self.max_energy)
+        axHist.legend(loc="best", title="counts after # steps")
+
+        plt.draw()
+        plt.show()
+
+    def plot_calorimeter(self):
+        """
+        Plots an imshow of the particle calorimeter data.
+        The data is summed over all energy bins and shown in units of keV.
+        """
+        # half the opening angles
+        half_yaw = self.opening_yaw / 2
+        half_pitch = self.opening_pitch / 2
+        # solid angle "view range" of the calorimeter detector
+        solid_angle_range = (-half_yaw + self.pos_yaw,
+                             half_yaw + self.pos_yaw,
+                             -half_pitch + self.pos_pitch,
+                             half_pitch + self.pos_pitch)
+
+        # unit conversion: Joules to eV
+        J2EV = 1.602e-19
+        # unit conversion: simUnit to EV
+        simUnitEV = self.sim_unit / J2EV
+
+        # data summed over the energy bins
+        calo_sum = np.sum(self.calo_data, axis=0) * simUnitEV / 1e3
+
+        figCal = plt.figure(1)
+        axCal = figCal.add_subplot(111)
+
+        caloPlot = axCal.imshow(
```

Comment:
INFO:
if you are implementing log scale in energy, you should not use imshow, since this fails with logscale above matplotlib >= 2.0.0